daemon/cmd/kube_proxy_replacement.go

```diff
@@ -45,10 +45,19 @@ import (
 func initKubeProxyReplacementOptions() error {
 	if option.Config.KubeProxyReplacement != option.KubeProxyReplacementStrict &&
 		option.Config.KubeProxyReplacement != option.KubeProxyReplacementPartial &&
-		option.Config.KubeProxyReplacement != option.KubeProxyReplacementDisabled {
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementDisabled &&
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementTrue &&
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementFalse {
 		return fmt.Errorf("Invalid value for --%s: %s", option.KubeProxyReplacement, option.Config.KubeProxyReplacement)
 	}
 
+	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementStrict ||
+		option.Config.KubeProxyReplacement == option.KubeProxyReplacementPartial ||
+		option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled {
+		log.Warnf("Deprecated value for --%s: %s (use either \"true\", or \"false\")", option.KubeProxyReplacement, option.Config.KubeProxyReplacement)
```

Comment:
@brb I think you forgot to update the ginkgo tests. This warning is now popping up in many of them.

Reply:
Thanks for reporting! https://github.com/cilium/cilium/pull/29695.